apps/meteor/tests/unit/app/livechat/server/hooks/beforeNewRoom.spec.ts

```diff
@@ -0,0 +1,21 @@
+import { expect } from 'chai';
+import proxyquire from 'proxyquire';
+
+import { callbacks } from '../../../../../../lib/callbacks';
+
+proxyquire.noCallThru().load('../../../../../../ee/app/livechat-enterprise/server/hooks/beforeNewRoom.ts', {
+	'meteor/meteor': {
+		Meteor: {
+			Error,
+		},
+	},
+});
+
+describe('livechat.beforeRoom', () => {
```

Comment:
Can you add more cases? What happens when extradata override a field from room? Also, when customFields is not an obj, etc